hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectSequenceInformationTest.java

```diff
@@ -0,0 +1,111 @@
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.tool.schema.extract.internal.SequenceInformationExtractorLegacyImpl;
+import org.hibernate.tool.schema.extract.spi.ExtractionContext;
+import org.hibernate.tool.schema.extract.spi.SequenceInformation;
+
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInHibernate;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * used driver hibernate.connection.driver_class com.microsoft.sqlserver.jdbc.SQLServerDriver
+ *
+ * @author Frank Doherty
+ */
+@RequiresDialect(value = { SQLServer2012Dialect.class })
+public class SQLServerDialectSequenceInformationTest extends BaseCoreFunctionalTestCase {
+
+	private final String DATABASE_NAME = "hibernate_orm_test_seq";
+
+	@Before
+	public void init() {
+		doInAutoCommit( "DROP DATABASE " + DATABASE_NAME );
+
+		// Latin1_General_CS_AS is a case-sensitive collation
+		doInAutoCommit( "CREATE DATABASE " + DATABASE_NAME + " COLLATE Latin1_General_CS_AS" );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-13141")
+	public void testExtractSequenceInformationForSqlServerWithCaseSensitiveCollation() {
+		String databaseNameToken = "databaseName=";
+		String url = (String) Environment.getProperties().get( AvailableSettings.URL );
+		String[] tokens = url.split( databaseNameToken );
+		String newUrl = tokens[0] + databaseNameToken + DATABASE_NAME;
+
+		Dialect dialect = serviceRegistry().getService( JdbcServices.class ).getDialect();
+
+		doInHibernate( this::sessionFactory, session -> {
+			StandardServiceRegistryBuilder ssrb = new StandardServiceRegistryBuilder();
+			ssrb.applySettings( Collections.singletonMap( AvailableSettings.URL, newUrl ) );
+			StandardServiceRegistry ssr = ssrb.build();
+
+			try ( Connection connection = ssr.getService( JdbcServices.class )
+					.getBootstrapJdbcConnectionAccess()
+					.obtainConnection() ) {
+				JdbcEnvironment jdbcEnvironment = new JdbcEnvironmentImpl( connection.getMetaData(), dialect );
+				Iterable<SequenceInformation> information = SequenceInformationExtractorLegacyImpl.INSTANCE.extractMetadata(
+						new ExtractionContext.EmptyExtractionContext() {
+							@Override
+							public Connection getJdbcConnection() {
+								return connection;
+							}
+
+							@Override
+							public JdbcEnvironment getJdbcEnvironment() {
+								return jdbcEnvironment;
+							}
+						} );
+				assertNotNull( information );
+			}
+			catch ( SQLException e ) {
+				log.error( e );
+				fail( "Sequence information was not retrieved: " + e.getMessage() );
+			}
+			finally {
+				StandardServiceRegistryBuilder.destroy( ssr );
+			}
+		} );
+	}
+
+	private void doInAutoCommit( String updateQuery ) {
```

Comment:
We have a `doInAutoCommit``method in `AbstractStoredProcedureTest`. I think we should move that to `TransactionUtil`.